### PR TITLE
Fix Incorrect Help Page and made tests around help pages more robust

### PR DESCRIPTION
**Pull request by fhembroff**
**Description**
This PR does two things,

**1:**

It modifies this testing method,
https://github.com/dockstore/dockstore-cli/blob/411f34dcbee710e796d830b2be3c122e54830fbf/dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java#L449
such that it now checks that all of the arguments given in `argv` are on the help page.

**2:**

It fixes the following help pages (trying out the toggle feature here, if they are causing issues let me know and I can post the before and after help pages in the comments),

<details>
<summary><b>dockstore plugin download --help</b></summary>

**Before:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ dockstore plugin download --help --script

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore plugin --help
       dockstore plugin [parameters] [command]

Description:
  Configure and debug plugins

Commands: 
  list
    List currently activated file provision plugins
  download
    Download default file provisioning plugins


---------------------------------------------


```
**After:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ java -jar dockstore-client/target/dockstore-client-1.14.0-SNAPSHOT-shaded.jar plugin download --help

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore plugin download --help
       dockstore plugin download [parameters]

Description:
  Download default file provisioning plugins



---------------------------------------------


```

</details>
<details>
<summary><b>dockstore plugin list --help</b></summary>

**Before:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ dockstore plugin list --help --script

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore plugin --help
       dockstore plugin [parameters] [command]

Description:
  Configure and debug plugins

Commands: 
  list
    List currently activated file provision plugins
  download
    Download default file provisioning plugins


---------------------------------------------


```
**After:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ java -jar dockstore-client/target/dockstore-client-1.14.0-SNAPSHOT-shaded.jar plugin list --help

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore plugin list --help
       dockstore plugin list [parameters]

Description:
  List currently activated file provision plugins



---------------------------------------------


```

</details>

The below were causing problems, as they were both saying that their usage was `dockstore verify`, the issue being, that there is no command called `dockstore verify`.

<details>
<summary><b>dockstore tool verify --help</b></summary>

**Before:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ dockstore tool verify --help --script

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore verify --help
       dockstore verify [parameters]

Description:
  Print cwltool runner dependencies

Required parameters:
  --trs-id <trs-id>                      TRS ID
  --version-id <version-id>              Version ID
  --file-path <file-path>                Path of the test JSON relative to the primary descriptor file
  --platform <platform>                  Platform to report on
  --metadata <metadata>                  Additional information on the verification (notes, explanation)
  --platform-version <platform-version>  Version of the platform to report on
  --descriptor-type <descriptor-type>    Descriptor Type (CWL, WDL, NFL)

Optional parameters:
  --unverify <unverify>                  Use flag to unverify
                                         Default: false

---------------------------------------------

```
**After:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ java -jar dockstore-client/target/dockstore-client-1.14.0-SNAPSHOT-shaded.jar tool verify --help

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore tool verify --help
       dockstore tool verify [parameters]

Description:
  Print cwltool runner dependencies

Required parameters:
  --trs-id <trs-id>                      TRS ID
  --version-id <version-id>              Version ID
  --file-path <file-path>                Path of the test JSON relative to the primary descriptor file
  --platform <platform>                  Platform to report on
  --metadata <metadata>                  Additional information on the verification (notes, explanation)
  --platform-version <platform-version>  Version of the platform to report on
  --descriptor-type <descriptor-type>    Descriptor Type (CWL, WDL, NFL)

Optional parameters:
  --unverify <unverify>                  Use flag to unverify
                                         Default: false

---------------------------------------------


```

</details>
<details>
<summary><b>dockstore workflow verify --help</b></summary>

**Before:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ dockstore workflow verify --help --script

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore verify --help
       dockstore verify [parameters]

Description:
  Print cwltool runner dependencies

Required parameters:
  --trs-id <trs-id>                      TRS ID
  --version-id <version-id>              Version ID
  --file-path <file-path>                Path of the test JSON relative to the primary descriptor file
  --platform <platform>                  Platform to report on
  --metadata <metadata>                  Additional information on the verification (notes, explanation)
  --platform-version <platform-version>  Version of the platform to report on
  --descriptor-type <descriptor-type>    Descriptor Type (CWL, WDL, NFL)

Optional parameters:
  --unverify <unverify>                  Use flag to unverify
                                         Default: false

---------------------------------------------


```
**After:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ java -jar dockstore-client/target/dockstore-client-1.14.0-SNAPSHOT-shaded.jar workflow verify --help

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore workflow verify --help
       dockstore workflow verify [parameters]

Description:
  Print cwltool runner dependencies

Required parameters:
  --trs-id <trs-id>                      TRS ID
  --version-id <version-id>              Version ID
  --file-path <file-path>                Path of the test JSON relative to the primary descriptor file
  --platform <platform>                  Platform to report on
  --metadata <metadata>                  Additional information on the verification (notes, explanation)
  --platform-version <platform-version>  Version of the platform to report on
  --descriptor-type <descriptor-type>    Descriptor Type (CWL, WDL, NFL)

Optional parameters:
  --unverify <unverify>                  Use flag to unverify
                                         Default: false

---------------------------------------------


```

</details>

Finally, I corrected the help page for `dockstore checker update_version`. As it was showing the help page for `dockstore checker version_tag` instead. After reading through the [code](https://github.com/dockstore/dockstore-cli/blob/411f34dcbee710e796d830b2be3c122e54830fbf/dockstore-client/src/main/java/io/dockstore/client/cli/CheckerClient.java#L367-L390) for `dockstore checker update_version` it appears that it gets the checker workflow for an entry and then passes it to the `version_tag` code (unfortunately there are no tests for this code, at least that I could find). So, I essentially made the help page for `dockstore checker update_version` the same as `dockstore workflow version_tag`.

<details>
<summary><b>dockstore checker update_version --help</b></summary>

**Before:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ dockstore checker update_version --help --script

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore checker version_tag --help
       dockstore checker version_tag [parameters]

Description:
  Update certain fields for a given checker version.

Required Parameters:
  --entry <entry>                                      Complete checker path in Dockstore (ex. quay.io/collaboratory/seqware-bwa-workflow)
  --name <name>                                        Name of the checker version.

Optional Parameters
  --workflow-path <workflow-path>                      Path to checker descriptor
  --hidden <true/false>                                Hide the tag from public viewing, default false

---------------------------------------------


```
**After:**
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ java -jar dockstore-client/target/dockstore-client-1.14.0-SNAPSHOT-shaded.jar checker update_version --help --script

 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore checker update_version --help
       dockstore checker update_version [parameters]

Updates a specific version of an existing checker workflow.

Required Parameters:
  --entry <entry>                                      Complete checker path in Dockstore (ex. quay.io/collaboratory/seqware-bwa-workflow)
  --name <name>                                        Name of the checker version.

Optional Parameters
  --workflow-path <workflow-path>                      Path to checker descriptor
  --hidden <true/false>                                Hide the tag from public viewing, default false

---------------------------------------------


```

</details>

**Review Instructions**
Ensure that my modified test runs, and that the help page for `dockstore checker update_version --help` is valid.

**Issue**
https://github.com/dockstore/dockstore/issues/5342
[DOCK-2318](https://ucsc-cgl.atlassian.net/browse/DOCK-2318)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `./mvnw clean install` 
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

[DOCK-2318]: https://ucsc-cgl.atlassian.net/browse/DOCK-2318?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java [hunk 1]
```diff
@@ -347,112 +347,129 @@ private void assertDepsCommandOutput() {
...
-        checkCommandForHelp(new String[] { CHECKER });
-        checkCommandForHelp(new String[] { CHECKER, DOWNLOAD, HELP });
-        checkCommandForHelp(new String[] { CHECKER, LAUNCH, HELP });
-        checkCommandForHelp(new String[] { CHECKER, ADD, HELP });
-        checkCommandForHelp(new String[] { CHECKER, UPDATE, HELP });
-        checkCommandForHelp(new String[] { CHECKER, UPDATE_VERSION, HELP });
-        checkCommandForHelp(new String[] { CHECKER, TEST_PARAMETER, HELP });
+        checkCommandForHelp(new String[] { TOOL, SEARCH }, false);
+        checkCommandForHelp(new String[] { TOOL, INFO }, false);
+        checkCommandForHelp(new String[] { TOOL, "CWL" }, false);
```
Comment (fhembroff) [inline dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java:352]: Made the change to upper-case, as upper case is what is given in the help pages.

I'll create a PR after this one is merged to move "CWL" and "WDL" to constants.

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java [hunk 2]
```diff
@@ -347,112 +347,129 @@ private void assertDepsCommandOutput() {
...
+            strings.add(HELP);
+        }
         strings.add(CONFIG);
         strings.add(TestUtility.getConfigFileLocation(true));
 
         Client.main(strings.toArray(new String[0]));
-        assertTrue(systemOutRule.getText().contains("Usage: dockstore"));
+        assertTrue(systemOutRule.getText().contains("Usage: dockstore"), "'Usage: dockstore' isn't being displayed, this probably means"
+                + "that no help page is being shown");
+        assertTrue(systemOutRule.getText().contains("dockstore " + join(" ", argv)), "`Usage: dockstore` is being displayed"
```
Comment (fhembroff) [inline dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java:471]: Doing two separate checks, as some help pages look like this,
```
fhembroff@LL6496-Fhembr:~/Documents/work/incorrect-help-pages/dockstore-cli$ dockstore tool convert cwl2yaml --help
 ____             _        _                 
|  _ \  ___   ___| | _____| |_ ___  _ __ ___ 
| | | |/ _ \ / __| |/ / __| __/ _ \| '__/ _ \
| |_| | (_) | (__|   <\__ \ || (_) | | |  __/
|____/ \___/ \___|_|\_\___/\__\___/|_|  \___|

---------------------------------------------
See https://www.dockstore.org for more information

Usage: dockstore tool convert --help
       dockstore tool convert cwl2json [parameters]
       dockstore tool convert cwl2yaml [parameters]
       dockstore tool convert wdl2json [parameters]
       dockstore tool convert entry2json [parameters]

Description:
  They allow you to convert between file representations.

---------------------------------------------


```

Comment (fhembroff) [reply dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java:471]: And although it displays the correct usage for the help page I requested, it is not on the line that begins with `Usage: `.

File: dockstore-client/src/main/java/io/dockstore/client/cli/CheckerClient.java [hunk 3]
```diff
@@ -366,23 +366,19 @@ private void testParameterChecker(List<String> args) {
...
-
             // Get entry from path
             Entry entry = getEntryFromPath(entryPath);
-
             // Get checker workflow
             Workflow checkerWorkflow = getCheckerWorkflowFromEntry(entry, true);
-
             // Update version
             if (entry != null && checkerWorkflow != null) {
-                // Readd entry path to call, but with checker workflow
```
Comment (svonworl) [inline dockstore-client/src/main/java/io/dockstore/client/cli/CheckerClient.java:382]: In honor of this change, I shall now queue up some tunes from my favorite '80s-hair-metal-band-with-a-similarly-misspelled-name (Ratt).

**Review by svonworl — APPROVED** [review]

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java [hunk 4]
```diff
@@ -347,112 +347,129 @@ private void assertDepsCommandOutput() {
...
-    private void checkCommandForHelp(String[] argv) throws IOException {
+    /**
+     * This method passes all the values in argv and --help if includeHelpFlag is true to Client.main and ensures
+     * that the output contains,
+     * 1) "Usage: dockstore", this ensures that some help page is being displayed
+     * 2) "dockstore" followed by the strings in argv, for example, if argv = {"test", "blue", "green"}, it would ensure
+     * that the output contained the string "dockstore test blue green"
+     *
+     * @param argv the arguments to pass to Client.main, do not pass "--help"
+     * @param includeHelpFlag If true, "--help" will be passed to Client.main after the arguments in argv
```
Comment (denis-yuen) [inline dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java:457]: Do you ever use this as a null? If not, would be simpler to use a `boolean`

File: dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java [hunk 5]
```diff
@@ -202,14 +204,29 @@ protected void versionTagHelp() {
...
         out("");
+        printRequiredParametersForVersionTag();
+        printHelpFooter();
+    }
+
+    private void printRequiredParametersForVersionTag() {
         out("Required Parameters:");
         out("  " + ENTRY + " <entry>                                      Complete " + getEntryType().toLowerCase()
-            + " path in Dockstore (ex. quay.io/collaboratory/seqware-bwa-workflow)");
+                + " path in Dockstore (ex. quay.io/collaboratory/seqware-bwa-workflow)");
```
Comment (denis-yuen) [inline dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java:214]: If this is a workflow, example should probably be a workflow ID example 
(like `github.com/collaboratory/seqware-bwa-workflow` above)

File: dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java (hunk 4 above)
Comment (fhembroff) [reply dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java:457]: Good point. I have changed it to `boolean`, thanks.

File: dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java (hunk 5 above)
Comment (fhembroff) [reply dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java]: I have changed it to `github.com/DataBiosphere/topmed-workflows/UM_variant_caller_wdl`, which is [this workflow](https://qa.dockstore.org/workflows/github.com/DataBiosphere/topmed-workflows/UM_variant_caller_wdl:1.32.0?tab=launch). Thanks for catching this.

**Pull request merged**